### ISSUE TITLE
Release gallery 16.1.2 

### ISSUE THREAD

**Issue by AlexAndBear** [owncloud/gallery#837]
# Release 16.1.2

### Reason
* Fix: Show proper name in file menu https://github.com/owncloud/gallery/pull/836
* Fix: Don't return files when accessing a share files drop https://github.com/owncloud/gallery/pull/839
* Fix: Unbreak gallery with gallery.cnf https://github.com/owncloud/gallery/pull/840
* Fix: Align password public protected link, with new Login-UI https://github.com/owncloud/gallery/pull/846

### Product approval 

- [ ] Verify transifex commits are in master: If a folder https://github.com/owncloud/<APPNAME>/tree/master/l10n exists, then https://drone.owncloud.com/owncloud/<APPNAME> should receive nightly '[tx] updated from transifex` ...
- [ ] Ping pmaier to check `info.xml` description and screenshot
- [ ] Get approval on translations from pmaier - send the date of the last tx commit from drone to Patrick

### QA

- [x] Create **QA** Milestone and set the description to $version
- [x] Move tickets from milestone `development` to `qa` (testplan and tickets mentioned in release ticket)
- [x] Create release branch with the format "release-$version"
- [ ] Create "$version Testplan" ([QA templates](https://github.com/owncloud/QA/tree/master/Server) or [Enterprise](https://github.com/owncloud/qa-enterprise/tree/master/Core)) -> stub to be filled in: https://github.com/owncloud/gallery/issues/847
- [ ] copy new/reopened issues from the previous release ticket as 'Issues to review' below.
- [x] check if [Non-normal Release Process](https://confluence.owncloud.com/display/OG/Apps+with+differences+in+release+process) applies, and adapt this checklist.

### Documentation

- [ ] Changes in the docs needed? If so create issue in owncloud/docs.

### Marketing

- [ ] Ping @owncloud/marketing
- [ ] Update https://confluence.owncloud.com/display/PROD/Upcoming+Releases
- [ ] Request a screenshot if the app is new or changed fundamentally

### Build

All actions to be done on the release branch from here:
- [x] Bump version in `info.xml` (no version suffix)
- [ ] Bump "sonar.projectVersion=" in `sonar-project.properties` (no version suffix)
- [x] Check php version dependency in `info.xml`
- [x] Check minimum ownCloud version dependency in `info.xml`
- [x] Change CHANGELOG.md 'Unreleased' to '$version' (today's date)

#### Beta/RC

- [x] Ping product owner to create github tag for `v${version}-rc1` on release branch (see https://confluence.owncloud.com/display/EN/Product+Owners+List)
- [x] Build and sign RC tarball from github tag `v${version}-rc1` (see handbook for how to build)
- [x] Upload as Github release using changelog entries
- [ ] Test and document in testplan + manual explorative testing:
    - [ ] If this is an enterprise app: Enabling the app must start the grace period
    - [ ] If this is the initial release of an app: Check that the minimum version constraint resolves correctly
    - [ ] Add section "Changlog Testing" using github releases text or CHANGELOG.md from release branch
    - [ ] Do changelog tests
    - [ ] Do regression tests
    - [ ] => Final "Go" from QA (set 'QA-ready' label) => else do another RC after fixing issues in the "release-$version" branch

#### Final

- [x] Check that no documentation issues are pending
- [x] Rename last RC tarball to final
- [x] Create github tag for final `v${version}` on release branch
- [x] Upload as Github release using changelog entries
- [x] Ping `#marketing` channel to coordinate publishing

### Publishing

- [x] Upload tarball to marketplace (see handbook in confluence for details)
- [x] Publish tarball for customers  - [old 10](https://customer.owncloud.com/owncloud/index.php/apps/files/?dir=%2FownCloud-Enterprise%2FownCloud-10), [old 10.0](https://customer.owncloud.com/owncloud/index.php/apps/files/?dir=%2FownCloud-Server%2FownCloud-10.0) and [new portal](https://portal.owncloud.com/apps/files/?dir=/Portal%20Data/All%20Account%20Data/ownCloud%20Enterprise%20Resources%20Data/Server&fileid=5661) (see [handbook](https://confluence.owncloud.com/display/OG/Releasing+Apps) for details)
- [x] Send mail to release-coordination with link to this ticket and to the changelog
- [x] Copy email to rocket chat `#updates` channel

### Post-release

- [x] Add a new 'Unreleased' section in CHANGELOG.md (and also update links at the bottom!)
- [x] Create PR to merge changes back to the `master` branch with description "Closes #XYZ" with the release ticket number
- [x] Rename QA milestone to `$version` and close the milestone
- [ ] Update https://confluence.owncloud.com/display/PROD/Upcoming+Releases
- [ ] Ensure [release template](https://github.com/owncloud/administration-internal/edit/master/tools/release/templates/server_app_release_template.md) is up to date
- [ ] Ensure the testplan template ([community](https://github.com/owncloud/QA/tree/master/Server) or [enterprise](https://github.com/owncloud/qa-enterprise/tree/master/Core)) is up to date
- [x] Close the testplan issue

**Comment by phil-davis** [comment]
@JanAckermann maybe the owncloud min-version should be updated in appinfo/info.xml to be 10.6 or 10.7 (whenever this menu dropdown thing was released) - since the changes may or may not work with older oC10 versions.

**Comment by AlexAndBear** [comment]
@phil-davis I don't see any compatibility issues to owncloud 10.2 maybe @jnweiger can confirm this in 'QA' 
Tested my self with 10.2, works as designed

**Comment by ho4ho** [comment]
Could also include fix by #840 ?

**Comment by AlexAndBear** [comment]
@phil-davis updated min-version to 10.8 with https://github.com/owncloud/gallery/pull/846
@ho4ho thanks, adjusted

**Comment by jnweiger** [comment]
@JanAckermann I've created branch release-16.1.2, bumped versions, and created an RC1. 
- There is no sonar-project.properties file in this repo. Should we add one?
- make dist needed some help:  https://github.com/owncloud/gallery/commit/8055138f88d1df2f7fd728ee92a24b661137d819
- make dist has several deprecation warnings:

```
  - Installing symfony/yaml (v2.8.52): Extracting archive
Package facebook/webdriver is abandoned, you should avoid using it. Use php-webdriver/webdriver instead.
Package flow/jsonpath is abandoned, you should avoid using it. Use softcreatr/jsonpath instead.
Package jakub-onderka/php-console-color is abandoned, you should avoid using it. Use php-parallel-lint/php-console-color instead.
Package jakub-onderka/php-console-highlighter is abandoned, you should avoid using it. Use php-parallel-lint/php-console-highlighter instead.
Package jakub-onderka/php-parallel-lint is abandoned, you should avoid using it. Use php-parallel-lint/php-parallel-lint instead.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
```

**Comment by phil-davis** [comment]
> There is no sonar-project.properties file in this repo. Should we add one?

PR #845 is waiting to be merged. That gets CI up-to-date, and also drops PHP 7.2 as part of that. So we were thinking to get this release out first, in a `release-16.1.2` branch, with `composer.json` dependencies at PHP 7.2 level. Then do the CI updates in `master`.

I  don't know what the real maintenance status of this app is nowadays. Do we require it to have things like SonarCloud set up?

**Comment by phil-davis** [comment]
Note: the release branch has been created and the release can proceed in that branch.

I have merged PR #845 to master now - that gets CI passing in master, but the CI does require dropping PHP 7.2 and bumping phpunit to major version 9. They are changes that are needed because of what has moved forward in core master recently. It does not effect the fact that the gallery app code in the release branch should work fine with a core 10.8.0 system running with PHP 7.2 7.3 or 7.4.

We just need to be aware that the CI in the release is known to fail because it is out-of-date. We can build the release tarball from the release branch, do the release manual test plan, and make the release. At the end we just need to sort out the merge-back of the release branch.
